core/src/main/java/org/apache/carbondata/core/indexstore/AbstractTableDataMap.java

```diff
@@ -26,28 +27,29 @@
  * DataMap at the table level, user can add any number of datamaps for one table. Depends
  * on the filter condition it can prune the blocklets.
  */
-public interface TableDataMap extends EventListener {
+public abstract class AbstractTableDataMap implements EventListener {
```

Comment:
Since it is abstract class better name as `AbstractTableDataMap` and one Segment can have multiple datamaps so `SegmentDataMap` may not justify the name